### ISSUE TITLE
Type annotations in javac type of object creation

### ISSUE THREAD

**Issue by t-rasmud** [typetools/checker-framework#2230]
I am trying to get the explicit annotations at constructor invocation sites which requires me to make a call to `TreeUtils.typeOf()` as shown [here](https://github.com/t-rasmud/checker-framework/commit/9fdde94fb19f9b6fd58de0c3e361d582daef52d2#diff-b997fe0effda8ef8635e7396f1861e64R1018).
However, printing out the results of this call produces ambiguous results.

Consider the test case below:
```
import org.checkerframework.checker.tainting.qual.*;

class ExtendsAndAnnotation {}

class IssueATFAmbiguity {
    void testTainted() {
        new @Tainted Object();
    }
}
```
Running the tainting checker on this example (`javac -processor tainting IssueATFAmbiguity.java`) along with the changes [here](https://github.com/t-rasmud/checker-framework/commit/9fdde94fb19f9b6fd58de0c3e361d582daef52d2#diff-b997fe0effda8ef8635e7396f1861e64R1018) prints the following line:
`... java.lang.Object`
But, if I comment out the line `class ExtendsAndAnnotation {}` in the test case, the same code prints out the following line:
`... (@org.checkerframework.checker.tainting.qual.Tainted :: java.lang.Object)`

**Comment by wmdietl** [comment]
Thanks for the error report!
`javacutil` contains low-level utilities that directly interact with javac. `TreeUtils.typeOf()` directly returns the type that javac maintains for a particular tree. Unfortunately, that type doesn't consistently contain type annotations.

I tried your change with the Java 9 version of the Checker Framework (available at github.com/eisop/checker-framework) and in that version javac consistently returns an un-annotated type.
Consistency relative to the presence of the unrelated class is good, but not having the annotation at all is not ideal.
I will check whether this issue has been fixed in jdk/jdk or file a new bug.

In general, you probably shouldn't use this method. Instead, use higher-level methods in `framework` that contain many work-arounds to javac issues.
